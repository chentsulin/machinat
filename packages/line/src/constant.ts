export const CHANNEL_REQUEST_GETTER = Symbol(
  'channel_request_getter.line.machinat'
);
export const BULK_REQUEST_GETTER = Symbol('bulk_request_getter.line.machinat');

/** @internal */
export const LINE = 'line' as const;

/** @internal */
export const PATH_REPLY = 'v2/bot/message/reply';

/** @internal */
export const PATH_PUSH = 'v2/bot/message/push';

/** @internal */
export const PATH_MULTICAST = 'v2/bot/message/multicast';

/** @internal */
export const PATH_RICHMENU = 'v2/bot/richmenu';

export enum LiffContextOs {
  Ios,
  Android,
  Web,
}
