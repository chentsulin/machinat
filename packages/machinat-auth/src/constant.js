// @flow
import { namedInterface } from '@machinat/core/service';
import type { ServerAuthorizer, AuthModuleConfigs } from './types';

export const STATE_COOKIE_KEY = 'machinat_auth_state';
export const TOKEN_COOKIE_KEY = 'machinat_auth_token';
export const SIGNATURE_COOKIE_KEY = 'machinat_auth_signature';
export const ERROR_COOKIE_KEY = 'machinat_auth_error';

export const AUTH_SERVER_AUTHORIZERS_I = namedInterface<
  ServerAuthorizer<any, any>[]
>({
  name: 'ServerAuthorizersList',
  multi: true,
});

export const AUTH_MODULE_CONFIGS_I = namedInterface<AuthModuleConfigs>({
  name: 'AuthModuleConfigs',
});
