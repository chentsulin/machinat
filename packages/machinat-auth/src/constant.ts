import { makeInterface } from '@machinat/core/service';
import type { ServerAuthorizer, AuthModuleConfigs } from './types';

export const STATE_COOKIE_KEY = 'machinat_auth_state';
export const TOKEN_COOKIE_KEY = 'machinat_auth_token';
export const SIGNATURE_COOKIE_KEY = 'machinat_auth_signature';
export const ERROR_COOKIE_KEY = 'machinat_auth_error';

export const SERVER_AUTHORIZERS_I = makeInterface<ServerAuthorizer<any, any>[]>(
  { name: 'ServerAuthorizersList', multi: true }
);

export const MODULE_CONFIGS_I = makeInterface<AuthModuleConfigs>({
  name: 'AuthModuleConfigs',
});
