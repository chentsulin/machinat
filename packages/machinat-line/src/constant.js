// @flow
export const LINE = 'line';
export const LINE_NAMESPACE = 'Line';

export const LINE_NATIVE_TYPE = Symbol.for('machinat.line.native');

export const ENTRY_GETTER = Symbol.for('machinat.line.segment.entry_getter');

export const PATH_REPLY = 'v2/bot/message/reply';
export const PATH_PUSH = 'v2/bot/message/push';
export const PATH_MULTICAST = 'v2/bot/message/multicast';
