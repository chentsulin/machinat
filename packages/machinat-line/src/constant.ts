export const LINE = 'line';

export const CHANNEL_API_CALL_GETTER = Symbol(
  'line.segment.channel_api_call_getter'
);
export const BULK_API_CALL_GETTER = Symbol('line.segment.bulk_api_call_getter');

export const PATH_REPLY = 'v2/bot/message/reply';
export const PATH_PUSH = 'v2/bot/message/push';
export const PATH_MULTICAST = 'v2/bot/message/multicast';
export const PATH_RICHMENU = 'v2/bot/richmenu';
