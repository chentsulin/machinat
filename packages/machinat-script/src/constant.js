// @flow
export const MACHINAT_SCRIPT_TYPE = Symbol.for('machinat.component.script');

export const SCRIPT_STATE_KEY = '$$machinat:script';
