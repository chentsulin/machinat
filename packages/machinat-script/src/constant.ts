import { makeInterface } from '@machinat/core/service';
import type { MachinatScript } from './types';

export const MACHINAT_SCRIPT_TYPE = Symbol.for('machinat.component.script');

export const SCRIPT_STATE_KEY = '$$machinat:script';

export const SCRIPT_LIBS_I = makeInterface<MachinatScript<any, any, any, any>>({
  name: 'ScriptLibsList',
  multi: true,
});
