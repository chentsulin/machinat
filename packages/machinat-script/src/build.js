// @flow
// @jsx Machinat.createElement
import Machinat from '@machinat/core';
import { container } from '@machinat/core/service';
import { MACHINAT_SCRIPT_TYPE } from './constant';
import ScriptProcessor from './processor';
import resolveScript from './resolve';
import compile from './compile';
import type { MachinatScript, ScriptNode } from './types';

type ScriptBuildOtions<Meta> = {
  meta: Meta,
};

const build = <Value, Input, ReturnValue, Meta>(
  scriptName: string,
  src: ScriptNode<Value, Input, ReturnValue>,
  options?: ScriptBuildOtions<Meta>
): MachinatScript<Value, Input, ReturnValue, Meta> => {
  const segments = resolveScript(src);
  const { entriesIndex, commands } = compile(segments, { scriptName });

  const script: MachinatScript<Value, Input, ReturnValue, Meta> = {
    $$typeof: MACHINAT_SCRIPT_TYPE,
    name: scriptName,
    entriesIndex,
    commands,
    meta: (options?.meta: any),
    Init: (null: any),
  };

  script.Init = container({ deps: [ScriptProcessor] })(
    (processor: ScriptProcessor) => async ({ channel, goto, vars }) => {
      const runtime = await processor.init(channel, script, {
        vars,
        goto,
      });
      const result = await runtime.run();

      return [
        result.content,
        <Machinat.Thunk effect={() => processor.save(runtime)} />,
      ];
    }
  );

  return script;
};

export default build;
