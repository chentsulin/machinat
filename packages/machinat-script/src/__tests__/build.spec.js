import Machinat from '@machinat/core';
import build from '../build';
import { MACHINAT_SCRIPT_TYPE } from '../constant';
import {
  IF,
  THEN,
  ELSE_IF,
  ELSE,
  WHILE,
  PROMPT,
  VARS,
  LABEL,
  CALL,
  RETURN,
} from '../keyword';

it('work', () => {
  const ChildScript = build(
    'ChildScript',
    <>
      {() => <dolore />}
      <PROMPT setter={(_, ctx) => ({ x: ctx.x })} key="childPrompt" />
    </>
  );

  const MyScript = build(
    'MyScript',
    <>
      <LABEL key="start" />
      {() => <b>Lorem</b>}

      <IF condition={() => false} key="if">
        <THEN>
          <WHILE condition={() => true}>
            <LABEL key="first" />
            {() => <i>ipsum</i>}
            <PROMPT key="ask_1" setter={(_, ctx) => ({ a: ctx.a })} />
          </WHILE>
        </THEN>
        <ELSE_IF condition={() => true}>
          <LABEL key="second" />
          {() => <dolor />}
          <PROMPT key="ask_2" setter={(_, ctx) => ({ b: ctx.b })} />
          <RETURN />
        </ELSE_IF>
        <ELSE>
          <LABEL key="third" />
          {() => 'sit amet,'}
          <CALL
            script={ChildScript}
            withVars={() => ({ foo: 'bar' })}
            goto="childPrompt"
            key="call_1"
          />
        </ELSE>
      </IF>

      <LABEL key="end" />
      <VARS set={_ => ({ foo: 'bar' })} />
      {() => 'ad minim veniam'}
    </>
  );

  expect(MyScript.name).toBe('MyScript');
  expect(MyScript.$$typeof).toBe(MACHINAT_SCRIPT_TYPE);
  expect(typeof MyScript.Init).toBe('function');

  expect(MyScript.commands).toMatchSnapshot();
  expect(MyScript.entryPointIndex).toMatchInlineSnapshot(`
    Map {
      "start" => 0,
      "third" => 3,
      "call_1" => 4,
      "first" => 7,
      "ask_1" => 8,
      "second" => 11,
      "ask_2" => 12,
      "end" => 15,
    }
  `);
});
