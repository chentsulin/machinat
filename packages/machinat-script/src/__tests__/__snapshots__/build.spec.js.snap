// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`built script object 1`] = `
Array [
  Object {
    "render": [Function],
    "type": "content",
  },
  Object {
    "condition": [Function],
    "isNot": false,
    "offset": 5,
    "type": "jump_cond",
  },
  Object {
    "condition": [Function],
    "isNot": false,
    "offset": 9,
    "type": "jump_cond",
  },
  Object {
    "render": [Function],
    "type": "content",
  },
  Object {
    "goto": "childPrompt",
    "key": "call_1",
    "script": Object {
      "$$typeof": Symbol(machinat.component.script),
      "Init": [Function],
      "commands": Array [
        Object {
          "render": [Function],
          "type": "content",
        },
        Object {
          "filter": [Function],
          "key": "childPrompt",
          "setter": [Function],
          "type": "prompt",
        },
      ],
      "entriesIndex": Map {
        "childPrompt" => 1,
      },
      "meta": Object {
        "foo": "baz",
      },
      "name": "ChildScript",
    },
    "setter": undefined,
    "type": "call",
    "withVars": [Function],
  },
  Object {
    "offset": 10,
    "type": "jump",
  },
  Object {
    "condition": [Function],
    "isNot": true,
    "offset": 4,
    "type": "jump_cond",
  },
  Object {
    "render": [Function],
    "type": "content",
  },
  Object {
    "filter": [Function],
    "key": "ask_1",
    "setter": [Function],
    "type": "prompt",
  },
  Object {
    "offset": -3,
    "type": "jump",
  },
  Object {
    "offset": 5,
    "type": "jump",
  },
  Object {
    "render": [Function],
    "type": "content",
  },
  Object {
    "filter": [Function],
    "key": "ask_2",
    "setter": [Function],
    "type": "prompt",
  },
  Object {
    "type": "return",
    "valueGetter": [Function],
  },
  Object {
    "offset": 1,
    "type": "jump",
  },
  Object {
    "setter": [Function],
    "type": "set_vars",
  },
  Object {
    "render": [Function],
    "type": "content",
  },
]
`;

exports[`built script object 3`] = `
Array [
  Object {
    "render": [Function],
    "type": "content",
  },
  Object {
    "filter": [Function],
    "key": "childPrompt",
    "setter": [Function],
    "type": "prompt",
  },
]
`;
